continuum/scenarios/transformation_incremental.py

```diff
@@ -0,0 +1,68 @@
+from typing import Callable, List
+
+import numpy as np
+
+from continuum.task_set import TaskSet
+from continuum.datasets import _ContinuumDataset
+from continuum.scenarios import InstanceIncremental
+
+from torchvision import transforms
+
+
+class TransformationIncremental(InstanceIncremental):
+    """Continual Loader, generating datasets for the consecutive tasks.
+
+    Scenario: Every task contains the same data with different transformation
+    It is a cheap way to create instance incremental scenarios.
+    Classic transformation incremental scenarios are "permutations" and "rotations"
+
+    :param cl_dataset: A continual dataset.
+    :param nb_tasks: Number of tasks in the continuum.
+    :param incremental_transformations: list of transformations to apply to specific tasks
+    :param base_transformations: List of transformation to apply to all tasks.
+    """
+
+    def __init__(
+            self,
+            cl_dataset: _ContinuumDataset,
+            nb_tasks: int,
+            incremental_transformations: List[List[Callable]],
+            base_transformations: List[Callable] = None
+    ):
+        super(TransformationIncremental, self).__init__(
+            cl_dataset=cl_dataset,
+            nb_tasks=nb_tasks,
+            base_transformations=base_transformations
+        )
+        if incremental_transformations is None:
+            raise ValueError("For this scenario a list transformation should be set")
+
+        if nb_tasks != len(incremental_transformations):
```

Comment:
We should remove the argument `nb_tasks` here because it is strictly equivalent in all cases to the number of transformations. So it can be inferred.

Reply:
I think it helps to avoid some errors to have it a bit redundant here. I would keep the nb_tasks parameter

Reply:
I disagree, the user interface should be simple with no redundance. Actually this redundance in the input arguments is source of bugs from the user code.

To avoid errors, it's on us to do internal checks.

Reply:
Ok, it makes sens here modif in beae858